lib/stat_tracker.rb

```diff
@@ -177,70 +153,71 @@ def worst_coach
   end
 
-  def calculate_team_accuracies(season_id, season_games)
+  # This is a helper method for most/least accurate team methods
+  def calculate_accuracies(season_id, season_games)
     team_goals = Hash.new { |hash, key| hash[key] = [0, 0] }
-  
     season_games.each do |game_team|
       team_goals[game_team.team_id][0] += game_team.goals
       team_goals[game_team.team_id][1] += game_team.shots
     end
-  
     team_accuracies = {}
-  
     team_goals.each do |team_id, (goals, shots)|
-      accuracy = shots.zero? ? 0 : (goals.to_f / shots.to_f) * 100
-      team_name = team_name_by_id(team_id)
+      accuracy = shots.zero? ? 0 : (goals / shots.to_f) * 100
+      team_name = name_by_id(team_id)
       team_accuracies[team_name] = accuracy.round(2)
     end
-  
     team_accuracies
   end
   
-
+  # This is a helper method for most/least accurate team methods
   def organize_games_by_season(season_id)
     games_in_season = @games.find_all {|game| game.season == season_id}
     season_game_ids = games_in_season.map {|game| game.game_id}
     @game_teams.find_all {|game| season_game_ids.include?(game.game_id)}
   end
 
-  def team_name_by_id(team_id)
+  # This is a helper method for calculate accuracies helper method
+  def name_by_id(team_id)
     team = @teams.find { |team| team.team_id == team_id }
     team.team_name if team
   end
   
   def most_tackles(season_id)
-    games_in_season = @game_teams.find_all do |game|
-      game.game_id[0..3] == season_id[0..3]
-    end
-    games_group_by_team = games_in_season.group_by { |team| team.team_id }
-    team_tackles = games_group_by_team.map do |team, tackles|
-    [team, tackles.map { |game| game.tackles}]
+    games_in_season = season_games(season_id)
+    games_by_team = games_in_season.group_by { |team| team.team_id }
+    team_tackles = games_by_team.map do |team, tackles|
+      [team, tackles.map { |game| game.tackles}]
     end
     tackle_totals = team_tackles.map {|team, tackles| [team, tackles.sum]}.to_h
     highest_total = tackle_totals.max_by { |team, tackles| tackles}
     @teams.find { |team| team.team_id == highest_total.first}.team_name
   end
 
   def fewest_tackles(season_id)
-    games_in_season = @game_teams.find_all do |game|
-      game.game_id[0..3] == season_id[0..3]
-    end
-    games_group_by_team = games_in_season.group_by { |team| team.team_id }
-    team_tackles = games_group_by_team.map do |team, tackles|
-    [team, tackles.map { |game| game.tackles}]
+    games_in_season = season_games(season_id)
+    games_by_team = games_in_season.group_by { |team| team.team_id }
+    team_tackles = games_by_team.map do |team, tackles|
+      [team, tackles.map { |game| game.tackles}]
     end
     tackle_totals = team_tackles.map {|team, tackles| [team, tackles.sum]}.to_h
     highest_total = tackle_totals.min_by { |team, tackles| tackles}
     @teams.find { |team| team.team_id == highest_total.first}.team_name
   end
```

Comment:
Prime candidate for refactor for sure.
I think that what I'm working on right now will also be able to benefit from this.